include/onnxruntime/core/framework/allocator.h

```diff
@@ -48,6 +48,9 @@ constexpr const char* OpenVINO_GPU = "OpenVINO_GPU";
 
 constexpr size_t kAllocAlignment = 256;
 
+class IAllocator;
+std::function<void*(size_t)> GetAllocationFn(std::shared_ptr<IAllocator> allocator, bool use_reserve, Stream* stream, WaitNotificationFn wait_fn);
```

Comment:
Why is the allocator aware of the stream and notification?

Reply:
This function is mainly used for GPU EP's GetScratch buffer, which need to aware which stream the scratch buffer is assigned to. it is a helper function to wrapper the stream aware allocation function, it won't impact other ep's usage.